styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/FlyteRunnerHandler.java

```diff
@@ -38,17 +49,46 @@
   public static final int STATIC_EXIT_CODE = 0;
 
   private static final Logger LOG = LoggerFactory.getLogger(FlyteRunnerHandler.class);
+  private final FlyteRunner flyteRunner;
+  private final Function<String, String> styxExecIdToFlyteNameMapper;
+
+  public FlyteRunnerHandler(FlyteRunner flyteRunner) {
+    this(flyteRunner, new StyxIdToFlyteExecNameMapper());
+  }
 
-  public FlyteRunnerHandler() {
+  @VisibleForTesting
+  FlyteRunnerHandler(FlyteRunner flyteRunner, Function<String, String> styxExecIdToFlyteNameMapper) {
     super(desc -> desc.flyteExecConf().isPresent());
+    this.flyteRunner = requireNonNull(flyteRunner);
+    this.styxExecIdToFlyteNameMapper = requireNonNull(styxExecIdToFlyteNameMapper);
   }
 
   @Override
   public void safeTransitionInto(final RunState state, final EventRouter eventRouter) {
     switch (state.state()) {
       case SUBMITTING:
         LOG.info("Entered state SUBMITTING for: " + state.workflowInstance());
-        final Event submitted = Event.submitted(state.workflowInstance(), state.data().executionId().orElseThrow(),
+
+        final FlyteExecConf flyteExecConf = state.data().executionDescription().orElseThrow().flyteExecConf().orElseThrow();
+        final String executionId = state.data().executionId().orElseThrow();
+        final String execName = styxExecIdToFlyteNameMapper.apply(executionId);
```

Comment:
Just currying around the styx-run-xxxx-... will make debugging difficult.
Should we instead create the FlyteExecutionId in a previous step?

Reply:
yes, maybe this could be converted [here](https://github.com/spotify/styx/blob/master/styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/ExecutionDescriptionHandler.java#L76) already?

Reply:
Let's add both to logs